hack/deployer/runner/tanzu.go

```diff
@@ -0,0 +1,420 @@
+func (t *TanzuDriver) create() error {
+	log.Println("Creating management cluster")
+	err := t.createInstallerConfig()
+	if err != nil {
+		return err
+	}
+	// run clean up and state upload to be able to continue to run operations later with a fresh Docker container
+	defer t.suspend()
+
+	cfgPathInContainer := filepath.Join("/root", TanzuInstallConfig)
+	if err := t.dockerizedTanzuCmd("management-cluster", "create", "--file", cfgPathInContainer).Run(); err != nil {
+		return err
+	}
+	log.Println("Creating workload cluster")
+	if err := t.dockerizedTanzuCmd("cluster", "create", t.plan.ClusterName, "--file", cfgPathInContainer).Run(); err != nil {
+		return err
+	}
+	return run([]func() error{
+		t.copyKubeconfig,
+		func() error {
+			return setupDisks(t.plan)
+		},
+		createStorageClass,
+	})
+}
+
+// createInstallerConfig renders the installer config into a file. This config can be used for both workload and management cluster
+// Tanzu ignores the worker machine count setting for management clusters.
+func (t *TanzuDriver) createInstallerConfig() error {
+	params := map[string]interface{}{
+		"AzureClientID":       t.azureCredentials.ClientID,
+		"AzureClientSecret":   t.azureCredentials.ClientSecret,
+		"AzureMachineType":    t.plan.MachineType,
+		"AzureLocation":       t.plan.Tanzu.Location,
+		"ResourceGroup":       t.plan.Tanzu.ResourceGroup,
+		"SSHPubKey":           t.plan.Tanzu.SSHPubKey,
+		"AzureSubscriptionID": t.azureCredentials.SubscriptionID,
+		"AzureTenantID":       t.azureCredentials.TenantID,
+		"NodeCount":           t.plan.Tanzu.NodeCount,
+	}
+	var cfg bytes.Buffer
+	if err := template.Must(template.New("tanzu-mgmt").Parse(TanzuInstallerConfigTemplate)).Execute(&cfg, params); err != nil {
+		return err
+	}
+
+	return ioutil.WriteFile(t.installerConfigFilePath(), cfg.Bytes(), 0600)
+}
+
+// installerConfigFilePath returns the path to the installer config valid in the context of deployer.
+func (t *TanzuDriver) installerConfigFilePath() string {
+	return filepath.Join(t.installerStateDir, TanzuInstallConfig)
+}
+
+// delete deletes the created resources simply by removing the Azure resource group
+func (t *TanzuDriver) delete() error {
+	return run([]func() error{
+		// shortcut deletion by not attempting deletion on application level but just delete the infrastructure
+		t.deleteResourceGroup,
+		t.deleteStorageContainer,
+		func() error {
+			user := fmt.Sprintf("%s-admin", t.plan.ClusterName)
+			context := fmt.Sprintf("%s@%s", user, t.plan.ClusterName)
+			return removeKubeconfig(context, t.plan.ClusterName, user)
+		},
+	})
+}
+
+// dockerizedTanzuCmd runs a tanzu CLI command inside a Docker container based off the Tanzu installer image.
+func (t *TanzuDriver) dockerizedTanzuCmd(args ...string) *Command {
+	params := map[string]interface{}{
+		"SharedVolume":        filepath.Join(SharedVolumeName(), filepath.Base(t.installerStateDir)),
```

Comment:
This doesn't work for me (Docker 19.03.13).
```
docker: Error response from daemon: create eck-ci-home-002657483/thb-dev-cluster2047199651: "eck-ci-home-002657483/thb-dev-cluster2047199651" includes invalid characters for a local volume name, only "[a-zA-Z0-9][a-zA-Z0-9_.-]" are allowed. If you intended to pass a host directory, use absolute path.
```

I don't think we can mount a given directory of a volume, we have to use the volume name.
```suggestion
		"SharedVolume":        SharedVolumeName(),
```

So, as an alternative we could make the `HOME` equals to `filepath.Join("\root", filepath.Base(t.installerStateDir))`?

Reply:
Yes that is the volume issue I mentioned above. Yes changing `$HOME` in the container might work, I also thought of just changing the bind mount for the case where we are not mounting a Docker volume. 

So in local mode (`make bootstrap-cloud`) we would use `$HOME/$installerStateDir` and in CI mode when we mount the named volume it would just be the root of that volume.

Reply:
I went with your proposal in the end @thbkrkr and changed the $HOME directory in the Tanzu container.